### DISCUSSION TITLE
Importing json on page load

### DISCUSSION THREAD

**Discussion by githubchicken** [williamtroup/Calendar.js#230] [General]
Hello. Just trying to import json from a file at page load. No errors but I'm missing something. Thanks for any help.

`<script>
        var calendarInstance = new calendarJs( "myCalendar", { 
            useLocalStorageForEvents: true,
            manualEditingEnabled: false, 
            views: {
                fullMonth: {
                    showTimesInEvents: false,
                    minimumDayHeight: 0
                }
            },
            manualEditingEnabled: false,
            visibleDays: [ 0, 1, 2, 3, 4, 5, 6 ],
        } );

        document.title += " v" + calendarInstance.getVersion();
        document.getElementById( "header" ).innerText += " - v" + calendarInstance.getVersion();
                
        function fetchJson() {
                fetch("http://localhost/hocal/events.json")
                .then((response) => response.json())
                .then((json) => console.log(json));
        }
        var myEvents = fetchJson();     
        calendarInstance.setEventsFromJson( myEvents );
        calendarInstance.addEventsFromJson( myEvents );
    </script>
`

Contents of events.json....

`[
  {
    "id":0,
    "title": "Marcus N",
    "from": "20240510",
    "to": "20240512",
    "color": "red"
  },
  {
    "id":1,
    "title": "Destinie",
    "from": "20240617",
    "to": "20240621",
    "color": "red"
  },
  {
    "id":2,
    "title": "Beth G",
    "from": "20231216",
    "to": "20231220",
    "color": "red"
  },
  {
    "id":3,
    "title": "Wendy R",
    "from": "20240621",
    "to": "20240623",
    "color": "red"
  }
]`

**Comment by reubenbrown13** [comment]
I am trying to do the same thing, and I am also not getting any data to populate.  What is strange is if I take the data that is getting returned, and put it in the "return" statement in my "getEvents" function, the data is populated.  It seems to be an issue with parsing the promise from fetch.  If anyone has a solution for this, it would be a big help.

**Comment by BradSpencer** [comment]
I am trying to make this work as well. I need to provide a picklist to users to select from which will populate the calendar based on the json file selected. This needs to be on the server so that many users can view it and can change it. When the export the events json file to the server it will be there for the next time a user selects the events file. I expect saving the json file to the server will be done with php (in my case)? Is this possible?
